src/home_robot/home_robot/motion/pinocchio_ik_solver.py

```diff
@@ -80,21 +80,35 @@ def compute_fk(self, q) -> Tuple[np.ndarray, np.ndarray]:
         return pos.copy(), quat.copy()
 
     def compute_ik(
-        self, pos: np.ndarray, quat: np.ndarray, q_init=None, max_iterations=100
+        self,
+        pos: np.ndarray,
+        quat: np.ndarray,
+        q_init=None,
+        max_iterations=100,
+        num_attempts: int = 1,
+        verbose: bool = False,
     ) -> Tuple[np.ndarray, bool]:
         """given end-effector position and quaternion, return joint values"""
         i = 0
         if q_init is None:
             q = self.q_neutral.copy()
+            if num_attempts > 1:
+                raise NotImplementedError(
+                    "Sampling multiple initial configs not yet supported by Pinocchio solver."
+                )
         else:
             q = self._qmap_control2model(q_init)
+            # Override the number of attempts
+            num_attempts = 1
         desired_ee_pose = pinocchio.SE3(R.from_quat(quat).as_matrix(), pos)
         while True:
             pinocchio.forwardKinematics(self.model, self.data, q)
             pinocchio.updateFramePlacement(self.model, self.data, self.ee_frame_idx)
 
             dMi = desired_ee_pose.actInv(self.data.oMf[self.ee_frame_idx])
             err = pinocchio.log(dMi).vector
+            if verbose:
+                print(i, err)
```

Comment:
Add more info to the printout if we want this to be a feature and not just a debug print.
Maybe `print("[pinocchio_ik_solver] iter={i}; error={err}")`

Reply:
done